app/controllers/posts_controller.rb

```diff
@@ -20,8 +20,7 @@ def show; end
 
   def index
     posts = current_user.followed_posts
-    sort_by = params[:sort_by]
-    @posts = define_posts(sort_by, posts)
+    @posts = Post.define_posts(posts, params)
```

Comment:
Let's avoid creating a method inside the Post model. Move the whole define_posts logic inside the controller.